src/compiler/scala/tools/nsc/transform/Mixin.scala

```diff
@@ -1037,4 +1053,14 @@ abstract class Mixin extends InfoTransform with ast.TreeDSL {
       finally localTyper = saved
     }
   }
+
+  private def isTraitMethodRequiringStaticImpl(dd: DefDef): Boolean = {
+    val sym = dd.symbol
+    sym != null && sym.owner.isTrait && !sym.isDeferred &&
+      (sym.isLazy || !sym.isAccessor) && !sym.isSuperAccessor && !sym.isModule && // these are not deferred, but defDef.rhs.isEmpty
```

Comment:
After the fields PR, we'll have SYNTHESIZE_IMPL_IN_SUBCLASS for accessors to denote their implementation is deferred to a subclass, but it's compiler-generated, so we should consider the method as concrete for name resolution etc. Super accessors don't get that flag yet, but maybe we should so that we can streamline this logic further (it also comes up in other spots).